File: C++/RenderGambar/Point.cpp
#include "Point.hpp"
#include <bits/stdc++.h>
using namespace std;
	
Point::Point(){             //ctor defult, absis = 0, ordinat = 0;
    absis = 0; 
    ordinat = 0;
}

//ctor user define
Point::Point(int newAbsis,int newOrdinat){
    absis = newAbsis;
    ordinat = newOrdinat;
}

//getter
int Point::getAbsis(){
    return absis;
}
int Point::getOrdinat(){
    return ordinat;
}

//setter
void Point::setAbsis(int newAbsis){
    absis = newAbsis;
}
void Point::setOrdinat(int newOrdinat){
    ordinat = newOrdinat;
}

void Point::geser(int deltaX, int deltaY){
    absis += deltaX;
    ordinat += deltaY;
}

int Point::hitungJarak(const Point& other){
    int dx = other.absis - absis;
    int dy = other.ordinat - ordinat;
    return sqrt(dx*dx + dy*dy);
}

File: Inttelijoopquarium/src/Point.java
import java.lang.Math;

public class Point implements Comparable {
  private int absis;
  private int ordinat;

  public Point() {
    absis = 0;
    ordinat = 0;
  }

  public Point(int absis, int ordinat) {
    this.absis = absis;
    this.ordinat = ordinat;
  }

  public int getAbsis() {
    return absis;
  }

  public void setAbsis(int absis) {
    this.absis = absis;
  }

  public int getOrdinat() {
    return ordinat;
  }

  public void setOrdinat(int ordinat) {
    this.ordinat = ordinat;
  }

  public double hitungJarak(Point point) {
    int dx = this.absis - point.absis;
    int dy = this.ordinat - point.ordinat;
    return Math.sqrt(dx * dx + dy * dy);
  }

  public void printInfo() {
    System.out.println("(" + absis + "," + ordinat + ")");
  }

  @Override
  public int compareTo(Object object) {
    Point point = (Point) object;
    if (absis == point.absis && ordinat == point.ordinat) {
      return 0;
    } else if (absis < point.absis && ordinat < point.ordinat) {
      return -1;
    } else {
      return 1;
    }
  }

  @Override
  public String toString() {
    return ("X = " + absis + " Y = " + ordinat);
  }
}